packages/decoder/lib/Web3Adapter.ts

```diff
@@ -0,0 +1,50 @@
+import Web3Eth from "web3-eth";
+type PastLogsOptions = {
+  toBlock?: string | number;
+  fromBlock?: string | number;
+  address?: string | string[];
+}
+import type BN from "bn.js";
+
+export class Web3Adapter {
+  public provider: any;
+  private eth: any;
+
+  constructor (provider: any) {
+    this.provider = provider;
+    // @ts-ignore - seems like the types are out of sync with the implementation
```

Comment:
you have an `any` **and** a `ts-ignore`!? happy to take a look with you if you had trouble getting better types working

Reply:
As I mentioned above, with web3, if you're not using `@types/web3`, then the types are indeed often out of sync with the implementation, unfortunately. :-/  (Unless that's fixed now, but I don't think it is?)  I'm assuming there's not a `@types/web3-eth` we can use?  Otherwise, uh... maybe you can manually type things...?